runtime-modules/working-group/src/lib.rs

```diff
@@ -384,6 +390,34 @@ decl_module! {
             Self::deposit_event(RawEvent::WorkerRoleAccountUpdated(worker_id, new_role_account_id));
         }
 
+        /// Update the associated role storage.
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn update_role_storage(
+            origin,
+            worker_id: WorkerId<T>,
+            storage: Vec<u8>
+        ) {
+            // Ensuring worker actually exists
+            let worker = Self::ensure_worker_exists(&worker_id)?;
+
+            // Ensure that origin is signed by member with given id.
+            ensure_on_wrapped_error!(
+                membership::Module::<T>::ensure_member_controller_account_signed(origin, &worker.member_id)
```

Comment:
I think for this extrinsic it makes more sense to authenticate origin is signed by the `worker.role_account_id`

Reply:
Done here https://github.com/Joystream/joystream/pull/2180/commits/7385ebfc75b1f10b59861054ae84d4c6a430cca1